python/ray/data/datasource/parquet_datasource.py

```diff
@@ -292,10 +242,6 @@ def __init__(
                 prefetch_remote_args[
                     "scheduling_strategy"
                 ] = DataContext.get_current().scheduling_strategy
-            if RETRY_EXCEPTIONS_FOR_META_FETCH_TASK is not None:
-                prefetch_remote_args[
-                    "retry_exceptions"
-                ] = RETRY_EXCEPTIONS_FOR_META_FETCH_TASK
```

Comment:
After thinking about it more, I think it'd better to still keep it. The realibility of metadata fetching is very critical, so it's no harm to have two levels of retries:
* retries inside the task itself
* retries the tasks here

Reply:
What would task-level retries achieve that Data-level retries don't? We're still performing task-level retries for system errors

Reply:
I think it'd better to perform task-level retries on this application error as well, given we have max attempts limit for task-internal-level retry. I want to make sure we are pretty reliable on this.

Reply:
Sounds like the primary advantage is that task-level retries don't have a limit, and Data-level retries do?

Reply:
we can also enforce task-level retries a limit. It's just have another layer of retry mechanism to be more safer. Suppose the node is bad, task-internal-retry won't have chance to retry on another node.

Reply:
> Suppose the node is bad, task-internal-retry won't have chance to retry on another node.

How would that happen? If the node is bad, that's a system error, and we retry system errors by default:

https://github.com/ray-project/ray/blob/ccfcb21c9ad9960604a349155300c55698e2e971/python/ray/data/_internal/remote_fn.py#L26

Reply:
> How would that happen? If the node is bad, that's a system error, and we retry system errors by default:

E.g. network is slow to keep having timeout. May not throw system error at all. It's kind of best practice in other systems to have retry on multiple layers - retry in same process, retry task, retry operator, retry job, etc.

Reply:
https://github.com/ray-project/ray/pull/42922/files#r1486855710